quinn-proto/src/token.rs

```diff
@@ -93,6 +98,22 @@ fn decode_addr<B: Buf>(buf: &mut B) -> Option<SocketAddr> {
     Some(SocketAddr::new(ip, port))
 }
 
+/// Reasons why a retry token might fail to validate a client's address
+#[derive(Debug, Copy, Clone)]
+pub(crate) enum RetryError {
```

Comment:
We have a `RetryError` now in `proto::endpoint` too, we should rename either one of these to help disambiguate?

Reply:
@Ralith ping, want to finish this up?

Reply:
Renamed this one `TokenDecodeError`. We can tolerate a bit more verbosity for an internal type, after all.